nucleus/dataset.py

```diff
@@ -94,28 +127,57 @@ def __eq__(self, other):
     @property
     def name(self) -> str:
         """User-defined name of the Dataset."""
-        return self.info().get(DATASET_NAME_KEY, "")
+        if self._name is None:
+            self._name = self._client.make_request(
+                {}, f"dataset/{self.id}/name", requests.get
+            )["name"]
+        return self._name
 
     @property
-    def model_runs(self) -> List[str]:
+    def model_runs(self) -> Dict[Any, Any]:
         """List of all model runs associated with the Dataset."""
         # TODO: model_runs -> models
-        return self.info().get(DATASET_MODEL_RUNS_KEY, [])
+        response = self._client.make_request(
+            {}, f"dataset/{self.id}/model_runs", requests.get
+        )
```

Comment:
Out of scope, but would be cool to start caching all of these responses locally as in `name` above!

Reply:
Put on the mental TODOs list. I think `slices` and `model_runs` warrant some extra looking into by using a `cachetools.TTLCache` such that we're fairly certain we would not serve stale information.